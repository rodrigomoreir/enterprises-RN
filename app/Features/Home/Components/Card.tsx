import React from 'react';
import { 
  TouchableOpacity,
  View, 
  Text,
  Image,
} from 'react-native';
import { getStyles } from './Card.style'

interface Props {
  title: string
  subtitle: string
  image: string
  onPress: () => void
}

const Card: React.FC<Props> = ({ title, subtitle, image, onPress }) => {
  const styles = getStyles()

  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.imageContainer}>
        <Image source={{uri: image}} style={styles.image} />
      </View>
      <View style={styles.textContainer}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.subtitle}>{subtitle}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default Card;
