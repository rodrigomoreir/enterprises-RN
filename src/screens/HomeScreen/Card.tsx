import React, { memo } from 'react'
import { Dimensions } from 'react-native'
import styled from 'styled-components/native'
import themeColors from '../../theme/theme'
import addOpacity from '../../utils/addOpacity'

interface Props {
  title: string
  subtitle: string
  image: string
  onPress: () => void
}

const screenWidth = Dimensions.get('window').width

const Card: React.FC<Props> = ({ title, subtitle, image, onPress }) => {
  return (
    <StyledTouchableOpacity onPress={onPress}>
      <StyledViewImage>
        <StyledImage source={{ uri: image }} />
      </StyledViewImage>
      <StyledViewText>
        <StyledTitle>{title}</StyledTitle>
        <StyledSubtitle>{subtitle}</StyledSubtitle>
      </StyledViewText>
    </StyledTouchableOpacity>
  )
}

const StyledTouchableOpacity = styled.TouchableOpacity`
  justify-content: center;
  height: 250px;
  width: ${screenWidth - 30}px;
  align-items: center;
  margin-top: 10px;
  margin-bottom: 10px;
  border-width: 1px;
  border-color: ${({ theme }) => theme.colors.grey};
  border-radius: 20px;
`
const StyledViewImage = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  border-bottom-width: 1px;
  border-bottom-color: ${({ theme }) => theme.colors.grey};
`
const StyledImage = styled.Image`
  height: 100%;
  width: 100%;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
`
const StyledViewText = styled.View`
  width: 100%;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 20px;
  background-color: ${addOpacity(themeColors.colors.blue, 0.4)};
`
const StyledTitle = styled.Text`
  align-self: center;
  font-size: 20px;
  line-height: 26.38px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
`
const StyledSubtitle = styled.Text`
  align-self: center;
  font-size: 15px;
  line-height: 26.38px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
`

export default memo(Card)
