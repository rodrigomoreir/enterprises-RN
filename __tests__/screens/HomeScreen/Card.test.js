import React from 'react'

// import renderer from 'react-test-renderer'
import { fireEvent } from '@testing-library/react-native'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../../src/functions/testing'

import Card from '../../../src/screens/HomeScreen/Card'

describe('Card', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(
      <Card
        title={'TEST'}
        subtitle={'SubTest'}
        image={''}
        onPress={jest.fn()}
      />,
    )
  })

  test('renders correctly', () => {
    const onPressMock = jest.fn()
    const { getByText } = renderWithTheme(
      <Card
        title={'TEST'}
        subtitle={'SubTest'}
        image={''}
        onPress={onPressMock}
      />,
    )
    const card = getByText('TEST')
    fireEvent.press(card)
    expect(onPressMock).toHaveBeenCalledTimes(1)
    expect(card).toBeTruthy()
  })
})
