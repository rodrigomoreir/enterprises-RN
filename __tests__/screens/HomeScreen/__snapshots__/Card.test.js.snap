// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#a9a9a9",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 250,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
      "opacity": 1,
      "width": 720,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#a9a9a9",
          "borderBottomWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Array [
          Object {
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#87cefa66",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#696969",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 26.38,
          },
        ]
      }
    >
      TEST
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#696969",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 26.38,
          },
        ]
      }
    >
      SubTest
    </Text>
  </View>
</View>
`;
